app/thirdparty/java/other/writeily/widget/WrMarkorWidgetProvider.java

```diff
@@ -53,67 +53,69 @@ public void onReceive(Context context, Intent intent) {
 
             // Open QuickNote
-            Intent openQuickNote = new Intent(context, OpenEditorFromShortcutOrWidgetActivity.class)
+            final Intent openQuickNote = new Intent(context, OpenEditorFromShortcutOrWidgetActivity.class)
                     .setAction(Intent.ACTION_EDIT)
                     .putExtra(Document.EXTRA_PATH, appSettings.getQuickNoteFile())
                     .putExtra(Document.EXTRA_PATH_IS_FOLDER, false);
-            views.setOnClickPendingIntent(R.id.widget_quicknote, PendingIntent.getActivity(context, requestCode++, openQuickNote, 0));
+            views.setOnClickPendingIntent(R.id.widget_quicknote, PendingIntent.getActivity(context, requestCode++, openQuickNote, PendingIntent.FLAG_UPDATE_CURRENT));
 
-            // Open Favourites
-            Intent openApp = new Intent(context, MainActivity.class).setAction(Intent.ACTION_VIEW);
-            views.setOnClickPendingIntent(R.id.widget_main, PendingIntent.getActivity(context, requestCode++, openApp, 0));
+            // Open Notebook
+            final Intent goHome = new Intent(context, MainActivity.class)
+                    .setAction(Intent.ACTION_VIEW)
+                    .putExtra(Document.EXTRA_PATH, appSettings.getNotebookDirectory())
+                    .putExtra(Document.EXTRA_PATH_IS_FOLDER, true);
+            views.setOnClickPendingIntent(R.id.widget_home, PendingIntent.getActivity(context, requestCode++, goHome, PendingIntent.FLAG_UPDATE_CURRENT));
 
             // ListView
-            Intent notesListIntent = new Intent(context, WrFilesWidgetService.class);
-            notesListIntent.putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, appWidgetId);
-            notesListIntent.putExtra(Document.EXTRA_PATH, directoryF);
-            notesListIntent.putExtra(Document.EXTRA_PATH_IS_FOLDER, true);
+            final Intent notesListIntent = new Intent(context, WrFilesWidgetService.class)
+                .putExtra(AppWidgetManager.EXTRA_APPWIDGET_ID, appWidgetId)
+                .putExtra(Document.EXTRA_PATH, directoryF)
+                .putExtra(Document.EXTRA_PATH_IS_FOLDER, true);
             notesListIntent.setData(Uri.parse(notesListIntent.toUri(Intent.URI_INTENT_SCHEME)));
 
             views.setEmptyView(R.id.widget_list_container, R.id.widget_empty_hint);
             views.setRemoteAdapter(R.id.widget_notes_list, notesListIntent);
 
-            Intent openNoteIntent = new Intent(context, OpenEditorFromShortcutOrWidgetActivity.class);
-            PendingIntent openNotePendingIntent = PendingIntent.getActivity(context, requestCode++,
-                    openNoteIntent, PendingIntent.FLAG_UPDATE_CURRENT);
+            final Intent openNoteIntent = new Intent(context, OpenEditorFromShortcutOrWidgetActivity.class);
+            final PendingIntent openNotePendingIntent = PendingIntent.getActivity(context, requestCode++, openNoteIntent, PendingIntent.FLAG_UPDATE_CURRENT);
```

Comment:
Refactor and improvements to the widget

1. Icons instead of unicode text on buttons
2. Tile opens widget folder
3. Home button opens notebook